python/tank/platform/validation.py

```diff
@@ -49,17 +49,23 @@ def validate_context(descriptor, context):
     """
     # check that the context contains all the info that the app needs
     context_check_ok = True
+
     req_ctx = descriptor.get_required_context()
-    for req_ctx_item in req_ctx:
-        context_check_ok &= (req_ctx_item == "user" and context.user is None)
-        context_check_ok &= (req_ctx_item == "entity" and context.entity is None)
-        context_check_ok &= (req_ctx_item == "project" and context.project is None)
-        context_check_ok &= (req_ctx_item == "step" and context.step is None)
-        context_check_ok &= (req_ctx_item == "task" and context.task is None)
+    if "user" in req_ctx and context.user is None:
+        context_check_ok = False
+    if "entity" in req_ctx and context.entity is None:
+        context_check_ok = False
+    if "project" in req_ctx and context.project is None:
+        context_check_ok = False
+    if "step" in req_ctx and context.step is None:
+        context_check_ok = False
+    if "task" in req_ctx and context.task is None:
+        context_check_ok = False
```

Comment:
i agree - i guess it was the C++ programmer in me that avoided that approach. I'll change it :)